### PR TITLE
Rich Text: Avoid activeElement focus call

### PR DESCRIPTION
**Pull request by aduth**
Fixes: https://core.trac.wordpress.org/ticket/49519

This pull request attempts to fix [Trac#49519](https://core.trac.wordpress.org/ticket/49519), where an error occurs when deleting blocks in the editor in Internet Explorer.

The error manifests itself as:

```
Unable to get property 'focus' of undefined or null reference
```

It was traced back to this line of code in the `applySelection` function:

https://github.com/WordPress/gutenberg/blob/11963546b7921f26d6f21a7b74e27ddfb9b8e1f4/packages/rich-text/src/to-dom.js#L305

This was recently changed/introduced as part of #19536.

It's not immediately clear to me whether this explicit focus is required. It may be that the manipulation of selections on the lines preceeding the `focus()` call would cause focus to be lost, so an explicit focus is intended in restoring the focus which had existed before those selection manipulations occurred.

Ultimately, it seems that `activeElement` is unset in Internet Explorer, but not other browsers.

If removing the `focus` is not acceptable, the following alternatives may be considered:

- Perform a truthy test of `activeElement`, to at least verify existence before calling `focus`
   - Why would this be any different or allowable in Internet Explorer? Is it a specific difference in browser implementation of `activeElement`? Specifically, I have found that `activeElement` will tend to fall back to `document.body` in many other browsers ([see related example](https://github.com/WordPress/gutenberg/blob/11963546b7921f26d6f21a7b74e27ddfb9b8e1f4/packages/components/src/guide/finish-button.js#L17)).
- Focus an element relative to the new selection, e.g. `range.commonAncestorContainer`

**Testing Instructions:**

In Internet Explorer and your preferred browser, verify no errors when deleting block:

1. Navigate to Posts > Add New
2. Insert a paragraph block with some text
3. Press <kbd>Enter</kbd> to create a second paragraph
4. Delete the second paragraph by pressing <kbd>Backspace</kbd>

### CODE REVIEW

**Comment by aduth** [comment]
> * Why would this be any different or allowable in Internet Explorer? Is it a specific difference in browser implementation of `activeElement`? Specifically, I have found that `activeElement` will tend to fall back to `document.body` in many other browsers ([see related example](https://github.com/WordPress/gutenberg/blob/11963546b7921f26d6f21a7b74e27ddfb9b8e1f4/packages/components/src/guide/finish-button.js#L17)).

Related (reinforcing):

>When there is no selection, the active element is the page's `<body>` or `null`.

https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/activeElement

**Review by jorgefilipecosta — APPROVED** [review]
Fixed the problem in my tests 👍 Block removal was possible on IE and other browsers. I verified the formatting library still works as before.

**Comment by aduth** [comment]
An additional finding from some last-minute testing: This resolves the error by avoiding calling on a property which may be `null` when there is no focused element. However, there is still a regression in WordPress in how selections are applied in Internet Explorer. Specifically, in Internet Explorer, when pressing <kbd>Backspace</kbd> from an empty second paragraph, the selection is placed at the _beginning_ of the first paragraph, instead of the end. It works correctly in other browsers. It would be good to investigate a fix here, though at least the error is resolved as of these changes. I expect the issue may be a result of the changes in #19536. I'll plan to open a follow-up issue.

**Comment by aduth** [comment]
>I'll plan to open a follow-up issue.

See: #20598

File: packages/rich-text/src/to-dom.js [hunk 1]
```diff
@@ -302,5 +302,24 @@ export function applySelection( { startPath, endPath }, current ) {
...
+		// element is not of the interface HTMLElement (does not have a `focus`
+		// or `blur` property).
+		//
+		// See: https://github.com/Microsoft/TypeScript/issues/5901#issuecomment-431649653
+		if ( activeElement ) {
+			if ( activeElement instanceof window.HTMLElement ) {
+				activeElement.focus();
+			}
+		} else if ( document.activeElement instanceof window.HTMLElement ) {
+			document.activeElement.blur();
```
Comment (ellatrix) [inline packages/rich-text/src/to-dom.js]: @aduth Why was this added? This seems to be causing the IE bug #20598.

Comment (aduth) [reply packages/rich-text/src/to-dom.js]: @ellatrix If I recall correctly, the idea was that if this logic should not cause a change in focus, and there was no focused element before the selection applied, we should guarantee that there is no focused element after it's applied by calling `blur` on whichever element is focused at that point.

I'm not able to remember if there was a specific case in mind this was encountering, or if it was only here to deal with the hypothetical scenario. 

Based on the debugging information from the original pull request comment, I suspect the main fix for [Trac#49519](https://core.trac.wordpress.org/ticket/49519) is largely in adding the condition of `if ( activeElement ) {` (accounting for the fact that `activeElement` can be `null`). If that's enough to accommodate all of the current scenarios and fix both [Trac#49519](https://core.trac.wordpress.org/ticket/49519) and #20598, then maybe it would be fine to remove this `blur` call.
